linetools/spectra/io.py

```diff
@@ -165,26 +163,37 @@ def readspec(specfil, inflg=None, efil=None, verbose=False, multi_ivar=False,
         print('Not sure what has been input.  Send to JXP.')
         return
 
-    if np.any(np.isnan(xspec1d.dispersion)):
+
+    # Generate, as needed
+    if 'xspec1d' not in locals():
+        # Give Ang as default
+        if not hasattr(wave, 'unit'):
+            uwave = u.Quantity(wave, unit=u.AA)
+        elif wave.unit is None:
+            uwave = u.Quantity(wave, unit=u.AA)
+        else:
+            uwave = u.Quantity(wave)
+        xspec1d = XSpectrum1D.from_tuple((wave, fx, sig, None))
+
```

Comment:
Why is this if block needed? When I refactored io.py I made sure all the read blocks returned an XSpectrum1D. As long as any new reading code does the same, this check shouldn't be necessary.  I also think the code is clearer without it - to me it's simpler to require that all the read sections/functions return an xspec1d, rather than have wave fx and sig variables floating around.